qt/python/mantidqtinterfaces/mantidqtinterfaces/simplescanviewer/rectangle_selection.py

```diff
@@ -0,0 +1,85 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import warnings
+
+from matplotlib.widgets import RectangleSelector
+
+
+class RectangleSelection(RectangleSelector):
+    """
+    This is a RectangleSelector that just overrides _release.
+    """
+
+    def _release(self, event):
+        """
+        Copyrights go to matplotlib, because this is a straight copy of the _release function of the RectangleSelector.
+        HOWEVER, this copied overriding function behaves differently than its original, for reasons I cannot explain.
+        Where the basic RectangleSelector stops reporting clicks (i.e. rectangle selectors with no width nor heights)
+        after one of those, this version continue, making it possible to switch between selectors in the view.
+
+        So for now, we use this version even if there is no rational reason I can see for it to be needed.
+        """
+
+        # matplotlib is bound to deprecate the eventpress attribute with the release of version 3.7, making it private
+        # In the meantime, it gives off warnings every time it is accessed.
+        # But since this feature does not work correctly at the moment, and we need this fix as a workaround, we just
```

Comment:
This is a bit worrying! I wonder if we can mitigate this somehow. @MathieuTillet Do you know if there's a matplotlib issue or more info about this bug anywhere that we could reference here so we can keep track of this?

@sf1919 Any thoughts? I believe we're pinned to matplotlib 3.5 at the moment but I can see us forgetting about this and being hit by problems when we update..!

Reply:
At present we are pinned to v3.5 and I don't see us removing a pin for matplotlib altogether. This means that we're unlikely to have a sudden failure relating to changes to matplotlib and should be able to handle changes as part of BAU. I'd like to think we wouldn't hit the major problems we did when changing to v3.5 earlier this year again because we'll we planning to upgrade more regularly.  

Going forward I think we're likely to raise issues for known deprecations (as @thomashampson has already done for some v3.5 ones we know we need to deal with in the next sprint) and work on those. When we're ready we'll open a PR and move the pin up to the next minor version (i.e. v3.6), see what falls over and fix accordingly before making the move to a new version.

I think if we know there is an upcoming problem with future deprecations we should be documenting this in an issue to keep track of it. That way we have more time to deal with it before we try and move the pin.

Reply:
@MathieuTillet Could you raise an issue so that we don't lose track of this, and add any more info you have about it there?

Reply:
#34684 for the deprecation warning issue itself.

As for the issue this entire class is solving, I have found no more info about it, but here is a short description for future reference:
on Ubuntu 18.04 at least, using matplotlib 3.5, the RectangleSelector doesn't let the user switch between selection. The signal for clicks stop being emitted after one.
For some reason, copy pasting the relevant code in a new class solved the issue.